document/core/text/types.rst

```diff
@@ -139,7 +154,16 @@ Memory Types
 .. math::
    \begin{array}{llclll@{\qquad\qquad}l}
    \production{memory type} & \Tmemtype &::=&
-     \X{lim}{:}\Tlimits &\Rightarrow& \X{lim} \\
+     \X{it}{:}\Tidxtype~~\X{lim}{:}\Tlimits &\Rightarrow& \X{it}~\X{lim} \\
+   \end{array}
+
+Abbreviations
+.............
+
```

Comment:
Add a short sentence here explaining the short-hand.

Reply:
Since abbreviations are supposed to be in-place expansions, doesn't this production need to be part of the main index type syntax rather than being an abbreviation?

Reply:
Are you saying I've not expressed the abbreviation correctly, or is there some reason why this type of abbreviation can't be expressed as an in-place expansion?

What I'm trying to say is specify just the `limits` expands to the `"i32" limits`.

Reply:
This can't be an in-place expansion because it would replace _every_ `limits` with `'i32' limits`. For example, if the text module contained `... i64 0 0 ...` it would expand to `... i64 i32 0 0 ...`. The expansion would never terminate and it would also produce an invalid module.

Reply:
I see, that makes sense there is nothing anchor the expansion.

Reply:
I think the expansion here should be one for index types, defined as `ε ≡ I32`.

Reply:
What kind of contexts
 are the expansions evaluated in? I thought we couldn’t do expansions in specific contexts like when we know an index type is expected.

Reply:
That is left a bit vague, intended to be applicable to a specific syntactic (which currently only appears in the Latex source :) ) and otherwise "clear from context". As mentioned in my SpecTec presentation last week, I'd like to remove the handwaviness and make this more precise, but that doesn't have super-high priority.

Reply:
Oh wow, that wasn't my interpretation at all. I thought you could have arbitrary `(param)` sprinkled in any and all locations throughout your wat and it would be ok because it would expand out to nothing before parsing. I'm glad to hear that's not the intention, though.